lib/iris/fileformats/netcdf.py

```diff
@@ -3206,3 +3245,114 @@ def is_valid_packspec(p):
 
         # Add conventions attribute.
         sman.update_global_attributes(Conventions=conventions)
+
+
+class ChunkControl(threading.local):
+    def __init__(self, var_dim_chunksizes=None):
+        """
+        Provide user control of Dask chunking.
+
+        The netcdf loader is controlled by the single instance of this : the
+        :data:`~iris.fileformats.netcdf.CHUNK_CONTROL` object.
+
+        A chunksize can be set for a specific (named) file dimension, when
+        loading specific (named) variables, or for all variables.
+
+        When a selected variable is a CF data-variable, which loads as a
+        cube, then the given dimension chunksize is *also* fixed for all
+        variables which are components of that cube, i.e. its Coords,
+        CellMeasures, Ancillary variables, etc.
+        This can be overriden, if required, by variable-specific settings.
+
+        For this purpose, Mesh coordinates and connectivities are *not* cube
+        components, and a chunk control on a cube data-variable will not affect
+        them.
+
+        """
+        self.var_dim_chunksizes = var_dim_chunksizes or {}
+
+    @contextmanager
+    def set(
+        self,
+        var_names: Union[str, Iterable[str]] = None,
+        **dimension_chunksizes: Mapping[str, int],
+    ) -> None:
+        """
+        Control the Dask chunksizes applied to netcdf variables during loading.
+
+        Parameters
+        ----------
+        var_names : str or list of str
+            apply the ``dimension_chunksizes`` controls only to these variables,
+            or when building cubes from these data variables.
+            If None (the default), settings apply to all loaded variables.
+        dimension_chunksizes : dict: str --> int
```

Comment:
TODO: this might be extended to more complex chunking controls, if useful?
Probably could do nonuniform sizes within a dimension (e.g. (3, 5, 5, 2)), but _not_ across multiple dims (not even sure how you say that, but I think is possible).
Plus None/auto/-1 as [already suggested](https://github.com/SciTools/iris/pull/4572#discussion_r810056992)